data_structures/src/builders.rs

```diff
@@ -0,0 +1,133 @@
+use std::time::{SystemTime, UNIX_EPOCH};
+use std::u32::MAX as U32_MAX;
+
+use rand::{thread_rng, Rng};
+
+use crate::types::{Address, Command, IpAddress, Message};
+
+////////////////////////////////////////////////////////////////////////////////////////
+// PROTOCOL MESSAGES CONSTANTS
+////////////////////////////////////////////////////////////////////////////////////////
+/// Magic number
+pub const MAGIC: u16 = 0xABCD;
+
+/// Version
+pub const VERSION: u32 = 0x0000_0001;
+
+/// Capabilities
+pub const CAPABILITIES: u64 = 0x0000_0000_0000_0001;
+
+/// User agent
+pub const USER_AGENT: &str = "full-node-desktop-edition";
+
+/// Genesis block
+pub const GENESIS: u64 = 0x0123_4567_89AB_CDEF;
+
+////////////////////////////////////////////////////////////////////////////////////////
+// BUILDER PUBLIC FUNCTIONS
+////////////////////////////////////////////////////////////////////////////////////////
+/// Function to build GetPeers messages
+pub fn build_get_peers() -> Message {
+    build_message(Command::GetPeers)
+}
+
+/// Function to build Peers messages
+pub fn build_peers(peers: &[SocketAddr]) -> Message {
+    // Cast all peers to witnet's address struct
+    let mut casted_peers = Vec::new();
+    peers.iter().for_each(|peer| {
+        casted_peers.push(to_address(*peer));
+    });
+
+    build_message(Command::Peers {
+        peers: casted_peers,
+    })
+}
+
+/// Function to build Ping messages
+pub fn build_ping() -> Message {
+    build_message(Command::Ping {
+        nonce: random_nonce(),
+    })
+}
+
+/// Function to build Pong messages
+pub fn build_pong(nonce: u64) -> Message {
+    build_message(Command::Pong { nonce })
+}
+
+/// Function to build Version messages
+pub fn build_version(
+    sender_addr: SocketAddr,
+    receiver_addr: SocketAddr,
+    last_epoch: u32,
+) -> Message {
+    build_message(Command::Version {
+        version: VERSION,
+        timestamp: current_timestamp(),
+        capabilities: CAPABILITIES,
+        sender_address: to_address(sender_addr),
+        receiver_address: to_address(receiver_addr),
+        user_agent: USER_AGENT.to_string(),
+        last_epoch,
+        genesis: GENESIS,
+        nonce: random_nonce(),
+    })
+}
+
+/// Function to build Verack messages
+pub fn build_verack() -> Message {
+    build_message(Command::Verack)
+}
+
+////////////////////////////////////////////////////////////////////////////////////////
+// AUX FUNCTIONS
+////////////////////////////////////////////////////////////////////////////////////////
+/// Function to build a message from a command
+fn build_message(command: Command) -> Message {
+    Message {
+        kind: command,
+        magic: MAGIC,
+    }
+}
+
+/// Function to get a random nonce
+fn random_nonce() -> u64 {
+    thread_rng().gen()
+}
+
+/// Function to get current timestamp (ms since Unix epoch)
+fn current_timestamp() -> u64 {
+    let now = SystemTime::now();
+    let now_duration = now.duration_since(UNIX_EPOCH).unwrap();
```

Comment:
I'm personally afraid of such kind of "brutely" unwrap everytime :smile: 
thus, how about ask help from `precise_time_ns -> u64` in the offical [rust-lang/time](https://docs.rs/time/0.1.40/time/fn.precise_time_ns.html) crate?

Reply:
I think that what we need in order to fill the timestamp field of the `Version` messages is a Unix timestamp (seconds or milliseconds since the Unix epoch). I see at the [`precise_time_ns`](https://docs.rs/time/0.1.40/time/fn.precise_time_ns.html) documentation that it uses an unspecified epoch so I'm not sure if we can use it.

I agree that the unwrap is dangerous.  I think that in the future this function may belong to the `EpochManager` which will detect if the system time is before the Unix epoch. If that happens, I think that it is ok the Witnet node fails. But anyway, I agree that we should control that error better.

Would it be ok if I returned a `Result` wrapping the timestamp?

Reply:
oh, sorry, my bad. timestamp should be a raw `u64`, `Result` mighty be unnecessarily complicated , so  just `.expect("err message")`  here?

Reply:
Perfect, I'll change it, thanks!

Reply:
The team told me that there is already a function in [`util`](https://github.com/witnet/witnet-rust/blob/master/util/src/timestamp.rs#L6) doing this, my bad :woman_facepalming: . @csmoe, would it be ok for you if I use the one that's already defined? It already wraps the error in a `WitnetResult` to be handled by the caller.

Reply:
@mmartinbar that's ok :)

Reply:
The team also told me to refactor the [`get_timestamp`](https://github.com/witnet/witnet-rust/blob/master/util/src/timestamp.rs#L6) function to use [**chrono**](https://docs.rs/chrono/0.4.0/chrono/struct.DateTime.html#method.timestamp).

Changes have been made across the code to change the timestamp type from u64 to i64 (flatbuffer schema changed, our own types as well...).